### PR TITLE
Psychotronics update and bugfixes.

### PR DESCRIPTION
**Pull request by Magisterium2022**
## Why It's Good For The Game

Fixes issue with Senior Mentalist, gives Mentalist role for assistance and future Psi-Research expansions.

## Changelog

add: Added Mentalist role, with Operant Psionics for Assay.
add: Gave Senior Mentalist anti-Psi weapon.
fix: Fixed bug from earlier PR which failed to give Senior Mentalist proper Psi rating.
/:cl:

### CODE REVIEW

**Comment by checkraisefold** [comment]
conflicts with #1767, we will see which one maintainers prefer

File: code/game/jobs/job/research.dm [hunk 1]
```diff
@@ -149,15 +149,65 @@
...
-		psi_faculties = list("[PSI_PSYCHOKINESIS]" = PSI_RANK_OPERANT)
-		psi_faculties = list("[PSI_ENERGISTICS]" = PSI_RANK_OPERANT)
+	if (H.mind?.role_alt_title == "Senior Mentalist") //Higher level of Psionics, as Seniormost expert on them in the site.
+		psi_faculties = list(
+			"[PSI_COERCION]" = PSI_RANK_MASTER,
+			"[PSI_PSYCHOKINESIS]" = PSI_RANK_OPERANT,
+			"[PSI_REDACTION]" = PSI_RANK_OPERANT,
+			"[PSI_ENERGISTICS]" = PSI_RANK_OPERANT
+			)
+	return ..()
```
Comment (checkraisefold) [inline code/game/jobs/job/research.dm:163]:
```suggestion
	. = ..()
	psi_faculties = list(
		"[PSI_COERCION]" = PSI_RANK_MASTER,
		"[PSI_PSYCHOKINESIS]" = PSI_RANK_OPERANT,
		"[PSI_REDACTION]" = PSI_RANK_OPERANT,
		"[PSI_ENERGISTICS]" = PSI_RANK_OPERANT
	)
```
sorry but locking it behind the alt title is genuinely dumb at this point
also style fixes

File: code/game/jobs/job/research.dm [hunk 2]
```diff
@@ -149,15 +149,65 @@
...
+	)
+
+	skill_points = 20
+	roleplay_difficulty = "Easy - Medium"
+	mechanical_difficulty = "Medium"
+
+/datum/job/mentalist/equip(mob/living/carbon/human/H)
+	if (H.mind?.role_alt_title == "Mentalist") //Basic level of Psionics, with ability to read others.
+		psi_faculties = list("[PSI_COERCION]" = PSI_RANK_OPERANT)
 	return ..()
```
Comment (checkraisefold) [inline code/game/jobs/job/research.dm:211]:
```suggestion
	. = ..()
	psi_faculties = list("[PSI_COERCION]" = PSI_RANK_OPERANT)
```

File: html/changelogs/Foundation-19-Magi.yml
Comment (checkraisefold) [inline html/changelogs/Foundation-19-Magi.yml:1]: remove this file. the changelog bot generates one for you from your PR description

File: code/game/jobs/job/research.dm
Comment (checkraisefold) [inline code/game/jobs/job/research.dm:1]: Senior Researcher has this:
```
alt_titles = list("Senior Xenobiologist", "Senior Xenoarcheologist", "Senior Xenobotanist", "Mentalist")
```
please remove the mentalist alt title, as you added a separate job for mentalist

File: code/game/jobs/job/research.dm [hunk 3]
```diff
@@ -149,15 +149,65 @@
...
+/datum/job/mentalist
+	title = "Psychotronics Researcher"
+	department = "Science"
+	department_flag = SCI
+	selection_color = "#633d63"
+	total_positions = 2
+	spawn_positions = 2
+	supervisors = "the Senior Psychotronics Researcher, Research Director, and Assistant Research Director"
+	economic_power = 4
+	requirements = list(EXP_TYPE_SCIENCE = 480)
```
Comment (checkraisefold) [inline code/game/jobs/job/research.dm:171]: upgrade the requirements for the senior psychotronics researcher. they are the same as this one.

File: code/game/jobs/job/research.dm [hunk 4]
```diff
@@ -149,15 +149,65 @@
...
+		ACCESS_SCIENCE_LVL3,
+		ACCESS_MEDICAL_LVL1,
+		ACCESS_RESEARCH
+	)
+	minimal_access = list()
+
+	min_skill = list(
+	    SKILL_CHEMISTRY	  = SKILL_TRAINED,
+		SKILL_MEDICAL	  = SKILL_TRAINED,
+	    SKILL_SCIENCE     = SKILL_EXPERIENCED
```
Comment (checkraisefold) [inline code/game/jobs/job/research.dm]:
```suggestion
	    SKILL_CHEMISTRY	  = SKILL_BASIC,
		SKILL_MEDICAL	  = SKILL_BASIC,
	    SKILL_SCIENCE     = SKILL_EXPERIENCED
```

**Review by checkraisefold — CHANGES_REQUESTED** [review]

File: code/game/jobs/job/research.dm (hunk 1 above)
Comment (checkraisefold) [inline code/game/jobs/job/research.dm:163]:
```suggestion
```

File: code/game/jobs/job/research.dm [hunk 5]
```diff
@@ -149,15 +149,65 @@
 	)
 
 	skill_points = 20
```
Comment (checkraisefold) [inline code/game/jobs/job/research.dm]:
```suggestion

	psi_faculties = list(
		"[PSI_COERCION]" = PSI_RANK_MASTER,
		"[PSI_PSYCHOKINESIS]" = PSI_RANK_OPERANT,
		"[PSI_REDACTION]" = PSI_RANK_OPERANT,
		"[PSI_ENERGISTICS]" = PSI_RANK_OPERANT
	)

	skill_points = 20
```

File: code/game/jobs/job/research.dm (hunk 2 above)
Comment (checkraisefold) [inline code/game/jobs/job/research.dm]:
```suggestion
```

File: code/game/jobs/job/research.dm [hunk 6]
```diff
@@ -149,15 +149,65 @@
...
+	    SKILL_SCIENCE     = SKILL_EXPERIENCED
+	)
+
+	max_skill = list(
+		SKILL_CHEMISTRY   = SKILL_MASTER,
+	    SKILL_MEDICAL     = SKILL_EXPERIENCED,
+	    SKILL_SCIENCE     = SKILL_MASTER
+	)
+
+	skill_points = 20
```
Comment (checkraisefold) [inline code/game/jobs/job/research.dm]:
```suggestion

	psi_faculties = list("[PSI_COERCION]" = PSI_RANK_OPERANT)
	skill_points = 20
```

**Review by checkraisefold — CHANGES_REQUESTED** [review]

**Comment by checkraisefold** [comment]
i dont like the alt titles controlling the psionic points, and it also causes them to change for anyone else joining the job later

File: code/game/jobs/job/research.dm (hunk 1 above)
Comment (checkraisefold) [inline code/game/jobs/job/research.dm]:
```suggestion
```

File: code/game/jobs/job/research.dm [hunk 7]
```diff
@@ -149,15 +149,65 @@
 	)
 
 	skill_points = 20
-	roleplay_difficulty = "Easy - Medium"
+	roleplay_difficulty = "Medium - Hard"
 	mechanical_difficulty = "Medium"
 
 /datum/job/seniormentalist/equip(mob/living/carbon/human/H)
```
Comment (checkraisefold) [inline code/game/jobs/job/research.dm]:
```suggestion
```

**Review by checkraisefold — CHANGES_REQUESTED** [review]
fixed my review

**Comment by Magisterium2022** [comment]
Thea is for the suggestions, and the discussion on Discord. I’ll see about implementing them in the morning, alongside some of your other suggestions.

**Comment by Magisterium2022** [comment]
> Thea is for the suggestions, and the discussion on Discord. I’ll see about implementing them in the morning, alongside some of your other suggestions.

Thanks, even.

**Comment by Magisterium2022** [comment]
> i dont like the alt titles controlling the psionic points, and it also causes them to change for anyone else joining the job later

I've just reworked that, you weren't the only one, and I agree with the points set forth in discussion.

File: code/game/jobs/job/research.dm (hunk 3 above)
Comment (Magisterium2022) [reply code/game/jobs/job/research.dm:171]: Updated to 620.

File: code/game/jobs/job/research.dm (hunk 4 above)
Comment (Magisterium2022) [reply code/game/jobs/job/research.dm]: Updated.

File: html/changelogs/Foundation-19-Magi.yml
Comment (Magisterium2022) [reply html/changelogs/Foundation-19-Magi.yml:1]: Removed.

File: code/game/jobs/job/research.dm [hunk 8]
```diff
@@ -149,16 +149,63 @@
...
-		psi_faculties = list("[PSI_COERCION]" = PSI_RANK_MASTER)
-		psi_faculties = list("[PSI_PSYCHOKINESIS]" = PSI_RANK_OPERANT)
-		psi_faculties = list("[PSI_ENERGISTICS]" = PSI_RANK_OPERANT)
-	return ..()
+	psi_faculties = list(
+		"[PSI_COERCION]" = PSI_RANK_OPERANT,
+		"[PSI_PSYCHOKINESIS]" = PSI_RANK_OPERANT,
+		"[PSI_REDACTION]" = PSI_RANK_MASTER,
+		"[PSI_ENERGISTICS]" = PSI_RANK_OPERANT
+		)
```
Comment (checkraisefold) [inline code/game/jobs/job/research.dm]:
```suggestion
```

File: code/game/jobs/job/research.dm [hunk 9]
```diff
@@ -149,16 +149,63 @@
 	)
 
 	skill_points = 20
-	roleplay_difficulty = "Easy - Medium"
+	roleplay_difficulty = "Medium - Hard"
 	mechanical_difficulty = "Medium"
 
 /datum/job/seniormentalist/equip(mob/living/carbon/human/H)
```
Comment (checkraisefold) [inline code/game/jobs/job/research.dm]:
```suggestion
```

File: code/game/jobs/job/research.dm [hunk 10]
```diff
@@ -149,16 +149,63 @@
 	)
 
 	skill_points = 20
```
Comment (checkraisefold) [inline code/game/jobs/job/research.dm:158]:
```suggestion
	psi_faculties = list(
		"[PSI_COERCION]" = PSI_RANK_OPERANT,
		"[PSI_PSYCHOKINESIS]" = PSI_RANK_OPERANT,
		"[PSI_REDACTION]" = PSI_RANK_MASTER,
		"[PSI_ENERGISTICS]" = PSI_RANK_OPERANT
	)

	skill_points = 20
```

File: code/game/jobs/job/research.dm [hunk 11]
```diff
@@ -149,16 +149,63 @@
...
+	    SKILL_SCIENCE     = SKILL_MASTER
+	)
+
+/datum/job/mentalist/equip(mob/living/carbon/human/H)
+	psi_faculties = list("[PSI_COERCION]" = PSI_RANK_OPERANT) //Basic level of Psionics, with ability to read others.
+
+	skill_points = 15
+	roleplay_difficulty = "Easy - Medium"
+	mechanical_difficulty = "Medium"
+
```
Comment (checkraisefold) [inline code/game/jobs/job/research.dm]:
```suggestion
	psi_faculties = list("[PSI_COERCION]" = PSI_RANK_OPERANT) //Basic level of Psionics, with ability to read others.
```

**Review by checkraisefold — CHANGES_REQUESTED** [review]
please directly commit my suggestions

File: code/game/jobs/job/research.dm (hunk 9 above)
Comment (Magisterium2022) [reply code/game/jobs/job/research.dm]: Directly committed all suggestions, as requested.

**Comment by Magisterium2022** [comment]
All requested changes have been directly committed.

File: code/game/jobs/job/research.dm [hunk 12]
```diff
@@ -148,17 +148,58 @@
...
+		"[PSI_REDACTION]" = PSI_RANK_MASTER,
+		"[PSI_ENERGISTICS]" = PSI_RANK_OPERANT
+	)
+
 	skill_points = 20
-	roleplay_difficulty = "Easy - Medium"
+	roleplay_difficulty = "Medium - Hard"
 	mechanical_difficulty = "Medium"
 
 /datum/job/seniormentalist/equip(mob/living/carbon/human/H)
```
Comment (checkraisefold) [inline code/game/jobs/job/research.dm]:
```suggestion
```

**Review by checkraisefold — CHANGES_REQUESTED** [review]

**Review by checkraisefold — APPROVED** [review]

File: code/game/jobs/job/research.dm [hunk 13]
```diff
@@ -148,17 +148,56 @@
 	    SKILL_SCIENCE     = SKILL_MASTER
 	)
 
+	psi_faculties = list(
+		"[PSI_COERCION]" = PSI_RANK_OPERANT,
+		"[PSI_PSYCHOKINESIS]" = PSI_RANK_OPERANT,
+		"[PSI_REDACTION]" = PSI_RANK_MASTER,
+		"[PSI_ENERGISTICS]" = PSI_RANK_OPERANT
+	)
+
```
Comment (checkraisefold) [inline code/game/jobs/job/research.dm]:
```suggestion
	psi_faculties = list(
		PSI_COERCION = PSI_RANK_OPERANT,
		PSI_PSYCHOKINESIS = PSI_RANK_OPERANT,
		PSI_REDACTION = PSI_RANK_MASTER,
		PSI_ENERGISTICS = PSI_RANK_OPERANT
	)

```

File: code/game/jobs/job/research.dm [hunk 14]
```diff
@@ -148,17 +148,56 @@
...
+		SKILL_MEDICAL	  = SKILL_BASIC,
+	    SKILL_SCIENCE     = SKILL_BASIC
+	)
+
+	max_skill = list(
+		SKILL_CHEMISTRY   = SKILL_EXPERIENCED,
+	    SKILL_MEDICAL     = SKILL_EXPERIENCED,
+	    SKILL_SCIENCE     = SKILL_MASTER
+	)
+	psi_faculties = list("[PSI_COERCION]" = PSI_RANK_OPERANT) //Basic level of Psionics, with ability to read others.
```
Comment (checkraisefold) [inline code/game/jobs/job/research.dm]:
```suggestion
	psi_faculties = list(PSI_COERCION = PSI_RANK_OPERANT) //Basic level of Psionics, with ability to read others.
```

**Review by checkraisefold — CHANGES_REQUESTED** [review]

**Review by checkraisefold — APPROVED** [review]

**Pull request merged**